docs/postmortems/2024-02-05-deprecated-icon-rendering.md

```diff
@@ -0,0 +1,113 @@
+---
+date: 2024-02-05
+authors: @literat
+---
+
+# Excessive deprecation logging while rendering Icon component
+
+**Summary**
+
+<!-- What is a one or two-line summary of the event that occurred? -->
+
+The `@lmc-eu/spirit-web-react@1.7.0` is shipped with a new deprecation warning about the `Icon` component.
+
+**Impact**
+
+<!-- What was the scope of impact from the event occurring? How many teams were
+impacted? -->
+
+The teams that use the ` @lmc-eu/spirit-web-react@1.7.0`` and above are impacted by this deprecation warning.
+ `Since `many components use` the `Icon` component, the impact should be extensive in logs that appear in logs both in browser developer tools and in the server logs.
+
+**Root causes**:
+
+<!-- Looking back, what ended up being the main reasons why this event occurred?
+-->
+
+These deprecations should solve the problem with server-side rendering in the Next.js framework where the Icons were not rendered properly. The deprecation warning was added to the `Icon` component to inform the developers that the `Icon` component will use an `html-react-parser`. A developer should install this dependency to make the `Icon` component work properly. This dependency will parse included SVG and render them directly to the DOM. After this, the usage of the `dangerouslySetInnerHTML` will be obsolete. For the condition of whether to use the `Icon` rendered by `html-react-parser` or the `dangerouslySetInnerHTML` was created a new utility called `htmlParser` dynamically imports the library and returns the function or nothing. This solution was not working properly. Also, the deprecation warning appears in both cases of the usage of the `Icon` component.
```

Comment:
```suggestion
These deprecations should solve the problem with server-side rendering in the Next.js framework where the Icons were not rendered properly. The deprecation warning was added to the `Icon` component to inform the developers that the `Icon` component will use the `html-react-parser`. A developer should install this dependency to make the `Icon` component work properly. This dependency will parse included SVG and render them directly to the DOM. After this, the usage of the `dangerouslySetInnerHTML` will be obsolete. For the condition of whether to use the `Icon` rendered by `html-react-parser` or the `dangerouslySetInnerHTML` a new utility called `htmlParser` was created which dynamically imports the library and returns the function or nothing. This solution was not working properly. Also, the deprecation warning appeared in both cases of the usage of the `Icon` component.
```

The last sentence is weird. As I understand it, it looks like the component was used only twice.

Reply:
Maybe generally split it into more sentences.

Reply:
Ok, fixed. Hope so...